### PR TITLE
Update Webhooks API to version 1.7

### PR DESCRIPTION
**Pull request by sdcoffey**

### CODE REVIEW

File: rest-api-sdk/src/main/java/com/paypal/api/payments/Webhook.java [hunk 1]
```diff
@@ -1,199 +1,186 @@
...
 import lombok.experimental.Accessors;
-import lombok.Getter; import lombok.Setter;
 
 import java.util.List;
 
-@Getter @Setter
+@Data
 @EqualsAndHashCode(callSuper = true)
 @Accessors(chain = true)
-public class Webhook  extends PayPalResource {
```
Comment (jaypatel512) [inline rest-api-sdk/src/main/java/com/paypal/api/payments/Webhook.java]: I think you accidently removed a lot of old methods on this one.

File: rest-api-sdk/src/main/java/com/paypal/api/payments/WebhookEvent.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.experimental.Accessors;
+
+import java.util.List;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+@Accessors(chain = true)
+public class WebhookEvent extends PayPalResource {
```
Comment (jaypatel512) [inline rest-api-sdk/src/main/java/com/paypal/api/payments/WebhookEvent.java]: This file already exists as `Event.java`.

File: rest-api-sdk/src/main/java/com/paypal/api/payments/WebhookEvent.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	 */
+	public WebhookEvent resend(APIContext apiContext) throws PayPalRESTException {
+		if (this.getId() == null) {
+			throw new IllegalArgumentException("Id cannot be null");
+		}
+
+		Object[] parameters = new Object[] {this.getId()};
+		String pattern = "v1/notifications/webhooks-events/{0}/resend";
+		String resourcePath = RESTUtil.formatURIPath(pattern, parameters);
+		String payLoad = this.toJSON();
```
Comment (jaypatel512) [inline rest-api-sdk/src/main/java/com/paypal/api/payments/WebhookEvent.java]: Resend does not have any body.

File: rest-api-sdk/src/main/java/com/paypal/api/payments/WebhookEventList.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.experimental.Accessors;
+
+import java.util.List;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+@Accessors(chain = true)
+public class WebhookEventList extends PayPalModel {
```
Comment (jaypatel512) [inline rest-api-sdk/src/main/java/com/paypal/api/payments/WebhookEventList.java]: This file is already present as `EventList.java`.

File: rest-api-sdk/src/main/java/com/paypal/api/payments/WebhookEventType.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+import com.paypal.base.rest.*;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.experimental.Accessors;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+@Accessors(chain = true)
+public class WebhookEventType extends PayPalResource {
```
Comment (jaypatel512) [inline rest-api-sdk/src/main/java/com/paypal/api/payments/WebhookEventType.java]: This file already exists as `EventType`.

File: rest-api-sdk/src/main/java/com/paypal/api/payments/WebhookEventTypeList.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.experimental.Accessors;
+
+import java.util.List;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+@Accessors(chain = true)
+public class WebhookEventTypeList extends PayPalModel {
```
Comment (jaypatel512) [inline rest-api-sdk/src/main/java/com/paypal/api/payments/WebhookEventTypeList.java]: This file already exists as `EventTypeList.java`.

**Comment by sdcoffey** [comment]
Ah ok. It's a little confusing going from language to language, as we have a `WebhookEvent` class in PHP. I'm going to close this and revisit this today - thanks for taking a peek!

File: rest-api-sdk/src/main/java/com/paypal/api/payments/Webhook.java (hunk 1 above)
Comment (sdcoffey) [inline rest-api-sdk/src/main/java/com/paypal/api/payments/Webhook.java]: I think all the same methods are still present, no? The only difference here is a that few methods taking an `accessToken` are deprecated.

**Comment by sdcoffey** [comment]
Also, re: `VerifyWebhookSignature`, I was under the impression that we didn't need/want to add this to SDKs that could perform the verification client-side, is that not the case?

**Comment by jaypatel512** [comment]
@sdcoffey ! You are correct. Just wanted to make sure we did that explicitly, or our SDK generator did not pick it up. 👍

File: rest-api-sdk/src/main/java/com/paypal/api/payments/Webhook.java (hunk 1 above)
Comment (sdcoffey) [inline rest-api-sdk/src/main/java/com/paypal/api/payments/Webhook.java]: Ah yeah it is a little weird that the diff shows and add/remove. :shake_fist: :github:

**Comment by sdcoffey** [comment]
Updated based on your comments, @jaypatel512 - let me know if you see anything else!

File: rest-api-sdk/src/main/java/com/paypal/api/payments/DefinitionsLinkdescription.java [hunk 7]
```diff
@@ -1,11 +1,11 @@
 package com.paypal.api.payments;
 
 import com.paypal.base.rest.PayPalModel;
+import lombok.Data;
 import lombok.EqualsAndHashCode;
 import lombok.experimental.Accessors;
-import lombok.Getter; import lombok.Setter;
 
-@Getter @Setter
```
Comment (jaypatel512) [inline rest-api-sdk/src/main/java/com/paypal/api/payments/DefinitionsLinkdescription.java]: We need to keep @Getter and @Setter for these classes. Using @Data causes `toString` to act weirdly.

File: rest-api-sdk/src/main/java/com/paypal/api/payments/Webhook.java [hunk 8]
```diff
@@ -1,199 +1,189 @@
...
 
 import com.paypal.base.rest.*;
+import lombok.Data;
 import lombok.EqualsAndHashCode;
 import lombok.experimental.Accessors;
-import lombok.Getter; import lombok.Setter;
 
 import java.util.List;
 
-@Getter @Setter
```
Comment (jaypatel512) [inline rest-api-sdk/src/main/java/com/paypal/api/payments/Webhook.java]: Can we keep @Getter and @Setter here instead of @Data.

**Pull request merged**